src/app/shared/components/multiselect-input/multiselect-input.component.ts

```diff
@@ -0,0 +1,75 @@
+import { Component, forwardRef, Input, OnInit } from '@angular/core';
+import {
+  ControlContainer,
+  AbstractControl,
+  ValidationErrors,
+  NG_VALIDATORS,
+  NG_VALUE_ACCESSOR,
+} from '@angular/forms';
+import { ListModel } from '../../models';
+
+@Component({
+  selector: 'app-multiselect-input',
+  templateUrl: './multiselect-input.component.html',
+  styleUrls: ['./multiselect-input.component.scss'],
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      multi: true,
+      useExisting: forwardRef(() => MultiselectInputComponent),
+    },
+    {
+      provide: NG_VALIDATORS,
+      multi: true,
+      useExisting: forwardRef(() => MultiselectInputComponent),
+    },
+  ],
+})
+export class MultiselectInputComponent implements OnInit {
```

Comment:
You have to implement ControlValueAccessor. And here you can use FormGroup.

Reply:
FormGroup? Not FormControl?

I tried to use FormGroup here with ControlValueAccessor, but there was a problem with "writeValue" method. As i found in google - it's because writeValue called before ngOnChange, and before ngOnChange this ng-select component is not ready to use. 

I made there some changes to call "patchValue" after ngOnChange, but there i had another errors with updating value of the form. 

I don't know why, but ... This multiselect we use to update "authors" field. And after updating it write to field "authors" parent object. And make it with recursion, so i got "infinite" nesting of form.authors.authors.authors...etc.

So finally i changed it to FormControl and it works fine. If you know why this happened - i'll change it to use FormGroup.